src/Controller/Rest/DocumentRestV2Controller.php

```diff
@@ -362,7 +366,7 @@ public function shareDocument(Request $request, AuthorizationCheckerInterface $a
                 $errors[] = 'User not found';
                 $status = Response::HTTP_BAD_REQUEST;
             } else {
-                $manager->generateSharedDocumentAndSendEmail($document, $email, $request->getLocale());
+                $manager->generateSharedPersonalDataAndSendEmail($document, $email, $request->getLocale());
```

Comment:
`generateSharedPersonalDataAndSendEmail` having the key word 'And' in the method name is often an indicator of single responsability principle violation. 
I think this controller method should first generate the personal data **then** send the email, calling two methods just as we do in `src/Controller/Rest/V3/DocumentApiController.php`